File: src/test/fixtures/cpp/coherence/related_classes.cpp
class CancellationToken;

class CancellationTokenSource {
public:
    CancellationToken token();
    void cancel() {}
};

class CancellationToken {
public:
    explicit CancellationToken(CancellationTokenSource source) {}
    bool isCancellationRequested() { return false; }
};

CancellationToken CancellationTokenSource::token() {
    return CancellationToken(*this);
}

File: src/test/fixtures/kotlin/coherence/RelatedClasses.kt
// decision: regression fixture for a real false positive - two classes whose methods
// construct/return each other must NOT be flagged as unrelated just because there are 2 of
// them in one file.

class CancellationTokenSource {
    fun cancel() {}

    fun token(): CancellationToken {
        return CancellationToken()
    }
}

class CancellationToken {
    fun linkedSource(): CancellationTokenSource {
        return CancellationTokenSource()
    }

    fun isCancellationRequested(): Boolean {
        return false
    }
}
